File: Java/PatchingArray.java
class Solution {
    public int minPatches(int[] nums, int n) {
        long right = 0;
        int patches = 0;

        for(int num=0; num<nums.length; num++){
            while((n > right) && (nums[num] > (right + 1))){
                right += right + 1;
                patches += 1;
            }
            right += nums[num];
        }
            
        while (n > right){
            right += right + 1;
            patches += 1;
        }    
        return patches;
    }
}

File: Python/PatchingArray.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        right = 0 
        patches = 0

        for num in nums:
            while n > right and num > right + 1:
                right += right + 1
                patches += 1
            right += num

        while n > right:
            right += right + 1
            patches += 1
            
        return patches 